src/uct/ib/rc/base/rc_iface.h

```diff
@@ -442,6 +460,49 @@ uct_rc_iface_handle_rndv_fin(uct_rc_iface_t *iface, struct ibv_exp_tmh *tmh)
     ucs_ptr_array_remove(&iface->tm.rndv_comps, ntohl(tmh->app_ctx), 0);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+uct_rc_iface_handle_rndv(uct_rc_iface_t *iface, struct ibv_exp_tmh *tmh,
+                         unsigned byte_len, unsigned flags)
+{
+    uct_ib_md_t *ib_md = uct_ib_iface_md(&iface->super);
+    struct ibv_exp_tmh_rvh *rvh;
+    unsigned tm_hdrs_len;
+    unsigned rndv_priv_hdr_len;
+    unsigned tmh_priv_data_len;
+    void *rndv_priv_hdr;
+    void *rb;
+    void *packed_rkey;
+
+    rvh = (struct ibv_exp_tmh_rvh*)(tmh + 1);
+
+    /* RC uses two headers: TMH + RVH, DC uses three: TMH + RVH + RAVH.
+     * So, get actual RNDV hdrs len from offsets */
+    tm_hdrs_len = sizeof(*tmh) +
+                  (iface->tm.rndv_desc.offset - iface->tm.eager_desc.offset);
+
+    rndv_priv_hdr     = (char*)tmh + tm_hdrs_len;
+    rndv_priv_hdr_len = byte_len - tm_hdrs_len;
+    tmh_priv_data_len = tmh->reserved[0];
+
+    if (tmh_priv_data_len) {
```

Comment:
this "if" is not really needed

Reply:
then we will invoke memcpy when it is not required (with 0 length though)

Reply:
i don't think it's a problem